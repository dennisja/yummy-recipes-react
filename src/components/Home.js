import React, { Component } from 'react';
import LoginForm from './forms/LoginForm';
import RegisterForm from './forms/RegisterForm';
import FormCard from './forms/FormCard';

class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
            loggedIn: false,
            userData: null,
        };
    }

    loginUser = (userData) => {
        this.setState({
            loggedIn: true,
            userData: userData,
        })
        console.log(userData)
    }

    render() {
        return (
            <div className="row">
                {
                    !this.state.loggedIn ?
                        (
                            <div>
                                <div className="col m6 hide-on-small-only">
                                    This is the demo
                                </div>
                                <div className="col m6 s10 offset-s1">
                                    <FormCard title="Login" form={<LoginForm onLoginSubmit={this.loginUser}/>} />
                                    <FormCard title="Register" form={<RegisterForm />} />
                                </div>
                            </div>
                        )
                        :
                        <div>I am loged in</div>
                }
            </div>
        );
    }
}

export default Home;
