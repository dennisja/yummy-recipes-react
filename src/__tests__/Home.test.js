import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Home from '../components/Home';

describe('Test Home component', () => {
    const homeWrapper = shallow(<Home location={{ state: { pathname: '/home' } }} loginUser={jest.fn()} loggedIn={false} />);
    it('renders properly', () => {
        expect(shallowToJson(homeWrapper)).toMatchSnapshot();
    });

    it('renders two FormCard components', () => {
        expect(homeWrapper.find('FormCard')).toHaveLength(2);
    });
});

