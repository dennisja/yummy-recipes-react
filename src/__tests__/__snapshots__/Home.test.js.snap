// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home component renders properly 1`] = `
<div
  className="row"
>
  <div
    className="col m6 hide-on-small-only"
  >
    This is the demo
  </div>
  <div
    className="col m6 s10 offset-s1"
  >
    <FormCard
      form={
        <LoginForm
          onLoginSubmit={[Function]}
        />
      }
      iconClass={null}
      title="Login"
    />
    <FormCard
      form={<RegisterForm />}
      iconClass={null}
      title="Register"
    />
  </div>
</div>
`;
